tests/integration/resources.bats

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	setup_busybox
+	update_config '.process.args = ["/bin/sh", "-c", "ulimit -n"]'
+}
+
+function teardown() {
+	teardown_bundle
+}
+
+@test "runc run with RLIMIT_NOFILE" {
+	update_config '.process.capabilities.bounding = ["CAP_RESOURCE"]'
```

Comment:
You probably meant `CAP_SYS_RESOURCE` here, but I am not sure why it is needed as we're running as root anyway.

Perhaps it's needed when running as rootless?

From resources(7):

<blockquote>
CAP_SYS_RESOURCE
* Use reserved space on ext2 filesystems;
* make ioctl(2) calls controlling ext3 journaling;
* override disk quota limits;
* increase resource limits (see setrlimit(2));
* override RLIMIT_NPROC resource limit;
* override maximum number of consoles on console allocation;
* override maximum number of keymaps;
* allow more than 64hz interrupts from the real-time clock;
* raise msg_qbytes limit for a System V message queue above the limit in /proc/sys/kernel/msgmnb (see msgop(2) and msgctl(2));
* allow the RLIMIT_NOFILE resource limit on the number of "in-flight" file descriptors to be bypassed when passing file descriptors to another process via a UNIX domain socket (see unix(7));
* override the /proc/sys/fs/pipe-size-max limit when setting the capacity of a pipe using the F_SETPIPE_SZ fcntl(2) command.
* use F_SETPIPE_SZ to increase the capacity of a pipe above the limit specified by /proc/sys/fs/pipe-max-size;
* override /proc/sys/fs/mqueue/queues_max limit when creating POSIX message queues (see mq_overview(7));
* employ the prctl(2) PR_SET_MM operation;
* set /proc/[pid]/oom_score_adj to a value lower than the value last set by a process with CAP_SYS_RESOURCE.
</blockquote>

Yes, "increase resource limits (see setrlimit(2))" is relevant, but the test can probably be rewritten to _decrease_ the RLIMIT_NOFILE.

Reply:
Thanks for your suggestion, I have modified it。
let me explain why CAP_SYS_RESOURCE is needed. Before calling exec, the container process has already applied seccomp and cap, so this process does not have permission to restore RLIMIT_NOFILE again. This will hide the problem, so I added CAP_SYS_RESOURCE permission to the container in the test case to ensure the problem be fixed.